### PR TITLE
test: Makes PullQueryLimitHARoutingTest more reliable

### PR DESCRIPTION
**Pull request by AlanConfluent**
### Description 
Previously, we weren't ensuring that the active was actually caught up with input, so refactored logic from RQTT to be used.   Also, we didn't ensure that the reported lags were zero, ensuring that standbys were caught up.

These introduced race conditions. where we would query a state store that wasn't guaranteed to be up to date with the upstream topics.

After discussing with @cprasad1, I decided to also remove the HA checks since we have these elsewhere and they have little to do with limit functionality.

Also, makes the test dedupe the results.  This is because after hitting the limit, the last row appears periodically (but rarely) to be duplicated.  I tried for a while to debug this, but this will have to wait for a future PR.

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/HighAvailabilityTestUtil.java [hunk 1]
```diff
@@ -352,6 +352,31 @@ static String extractQueryId(final String outputString) {
     };
   }
 
+  // Ensures that lags have been reported for the given host, but that the value is zero.
+  // Makes the simplified assumption that there's just one state store.
```
Comment (nateab) [inline ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/HighAvailabilityTestUtil.java:356]: im trying to think of a case where we might have more than one state store in a test, but i guess for `PullQueryLimitHARouting` we dont have any

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/IntegrationTestUtil.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+import org.apache.kafka.common.IsolationLevel;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.streams.KafkaStreams.State;
+import org.apache.kafka.streams.TopologyDescription;
+import org.apache.kafka.streams.TopologyDescription.Source;
+import org.apache.kafka.streams.TopologyDescription.Subtopology;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IntegrationTestUtil {
```
Comment (nateab) [inline ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/IntegrationTestUtil.java:26]: you just moved this logic right, i couldnt find any differences in the logic itself from a quick look

**Review by nateab — APPROVED** [review]
LGTM!

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/HighAvailabilityTestUtil.java (hunk 1 above)
Comment (AlanConfluent) [reply ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/HighAvailabilityTestUtil.java:356]: If we had multiple tables defined, it's possible.  But none of the tests which use this have that, so it's works fine.

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/IntegrationTestUtil.java (hunk 2 above)
Comment (AlanConfluent) [reply ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/IntegrationTestUtil.java:26]: Yeah, just cut and pasted it.  No intended changes.

**Comment by cprasad1** [comment]
LGTM @AlanConfluent . Can we rename the test as well to remove mention of HA?

**Review by cprasad1 — APPROVED** [review]

**Pull request merged**